lib/logstash/inputs/gelf.rb

```diff
@@ -96,7 +96,21 @@ def udp_listener(output_queue)
       # Gelfd parser outputs null if it received and cached a non-final chunk
       next if data.nil?
 
-      event = LogStash::Event.new(LogStash::Json.load(data))
+      begin
+        event = LogStash::Event.new(LogStash::Json.load(data))
+      rescue LogStash::Json::ParserError => e
+        @logger.info("JSON parse failure. Falling back to plain-text", :error => e, :data => data)
+        event = LogStash::Event.new("message" => data, "tags" => ["_jsonparsefailure"])
+      rescue StandardError => e
```

Comment:
this is how json_lines does it:
```ruby
  def legacy_parse(json, &block)
    # ignore empty/blank lines which LogStash::Json#load returns as nil
    o = LogStash::Json.load(json)
    yield(LogStash::Event.new(o)) if o
  rescue LogStash::Json::ParserError
    yield LogStash::Event.new("message" => json, "tags" => ["_jsonparsefailure"])
  end
```

and yes, it will address #26, all exceptions reported are `LogStash::Json::ParserError`.